pkg/provider/cloud/vsphere/provider.go

```diff
@@ -124,6 +126,18 @@ func (v *VSphere) reconcileCluster(ctx context.Context, cluster *kubermaticv1.Cl
 		}
 	}
 
+	if managedTags != nil {
+		if cluster.Status.ProviderStatus == nil {
+			cluster.Status.ProviderStatus = &kubermaticv1.ClusterProviderStatus{
+				VSphere: &kubermaticv1.VSphereStatus{},
+			}
+		}
+		if cluster.Status.ProviderStatus.VSphere == nil {
+			cluster.Status.ProviderStatus.VSphere = &kubermaticv1.VSphereStatus{}
+		}
+		cluster.Status.ProviderStatus.VSphere.ManagedTagIDs = sets.List(managedTags)
+	}
```

Comment:
These initialisations should only happen once. We are doing this in https://github.com/kubermatic/kubermatic/pull/14936/files#diff-c3a3b708946ae2d22a1525aaf944dec6a202e2a91a99b9e06b8cbe5d5cf65252R52-R63 as well. Why do we need this here as well?